homeassistant/components/tuya/select.py

```diff
@@ -411,32 +422,60 @@ def __init__(
     ) -> None:
         """Init Tuya sensor."""
         super().__init__(device, device_manager)
+
+        description_key = description.key
         self.entity_description = description
-        self._attr_unique_id = f"{super().unique_id}{description.key}"
+        self._attr_unique_id = f"{super().unique_id}{description_key}"
 
         self._attr_options: list[str] = []
+        self._dptype: DPType | None = None
+
         if enum_type := self.find_dpcode(
-            description.key, dptype=DPType.ENUM, prefer_function=True
+            description_key, dptype=DPType.ENUM, prefer_function=True
         ):
             self._attr_options = enum_type.range
+            self._dptype = DPType.ENUM
+
+        elif self.find_dpcode(
+            description_key, dptype=DPType.BOOLEAN, prefer_function=True
+        ):
+            self._attr_options = [self.to_value(True), self.to_value(False)]
```

Comment:
We can just hardcode this? It is now hardcoded as well, so it will always be the same?

```suggestion
            self._attr_options = ["true", "false"]
```